src/Core/System/SalesChannel/Entity/SalesChannelRepository.php

```diff
@@ -217,6 +217,9 @@ private function doSearch(Criteria $criteria, SalesChannelContext $salesChannelC
 
     private function processCriteria(Criteria $topCriteria, SalesChannelContext $salesChannelContext): void
     {
+        $event = new SalesChannelRepositoryProcessCriteriaEvent($this->definition, $topCriteria, $salesChannelContext);
```

Comment:
doesn't it make more sense to throw the event after the criteria was processed here? 

That way you get the exact critieria that will be used for the search, and you can manipulate it in the event and it won't be changed afterward